config/initializers/warden.rb

```diff
@@ -8,8 +8,9 @@
 
 require 'warden/strategies/basic_strategy'
 require 'warden/strategies/session_strategy'
+require "warden/strategies/doorkeeper"
 
 Rails.application.config.middleware.insert_after(ActionDispatch::Flash, Warden::Manager) do |manager|
-  manager.default_strategies :basic, :session
+  manager.default_strategies :doorkeeper, :basic, :session
```

Comment:
the other api PR had a solution for this

On Tue, Jul 26, 2016 at 7:47 PM, Adam Panzer <notifications@github.com>
wrote:

> In config/initializers/warden.rb
> <https://github.com/zendesk/samson/pull/1163#discussion_r72373176>:
>
> >
> >  Rails.application.config.middleware.insert_after(ActionDispatch::Flash, Warden::Manager) do |manager|
> > -  manager.default_strategies :basic, :session
> > +  manager.default_strategies :doorkeeper, :basic, :session
>
> We want doorkeeper to only work on api urls.
>
> I can look into ways of blocking it otherwise.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1163/files/ce06ab8ee433a5a4a031008d1931a7f15e0e5229#r72373176>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ1MYaGhekfsF-Gr_4jYuTpQcIC76ks5qZsa_gaJpZM4JVi_g>
> .
>